### ISSUE TITLE
Hosted Fields Instance focus is not working properly on iOS 13

### ISSUE THREAD

**Issue by EmilsK** [braintree/braintree-web#460]
### General information

* SDK version: 3.52.1
* Environment: Sandbox and Production
* Browser and OS: iPhone XS | Safari | iOS 13 
"Mozilla/5.0 (iPhone; CPU iPhone OS 13_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0 Mobile/15E148 Safari/604.1"

### Issue description

Hi, it looks like there is an issue with autofocus on new Safari in iOS 13.

What we have implemented is that when the user fills out the credit card number field and it is determined as valid it immediately changes the focus field to next one (ex. date, then from date to CVV).

We have successfully accomplished that and it was working as expected on previous iOS versions and desktops. 

Here are the code parts that are responsible for this: 

Calling a function makeFieldValid() on validity change:

```
hostedFieldsInstance.on('validityChange', function(event) {
    var field = event.fields[event.emittedBy];

    if (!field.isValid && !field.isPotentiallyValid) {
        // Case when there is an error
        makeFieldInvalid(event.emittedBy);
    } else if (!field.isValid && field.isPotentiallyValid) {
        // Case when it's neutral
        makeFieldNeutral(event.emittedBy);
    } else {
        // Case when it's valid
        makeFieldValid(event.emittedBy);
    }
});
```

makeFieldValid() function sets focus to next field if determined as valid:
```
function makeFieldValid(field) {
    var $fieldContainer = $(hostedFieldsInstance.getState().fields[field].container);

    $fieldContainer.next().html(getFieldLabel(field));
    $fieldContainer.next().css('color', 'green');
    $fieldContainer.css('border', '1px solid green');

    if (field == 'number') {
            hostedFieldsInstance.focus('expirationDate');
    } else if (field == 'expirationDate') {
            hostedFieldsInstance.focus('cvv');
    }
}
```
And here’s the on focus function for hosted field:
```
hostedFieldsInstance.on('focus', function(event) {
    var field = event.fields[event.emittedBy];
    $(field.container).next().addClass('focus');
});
```

When hostedFieldsInstance.focus is called the focus field is not changed although the on focus function is still called which can be seen when the field label changes its position and also testing with breakpoints in dev console shows the same.

Interesting is that it works as expected if "next" field has been in focus mode before, but not on the first time when that is necessary.

**Comment by crookedneighbor** [comment]
Thanks for the report, I was able to reproduce the issue on iOS 13.

iOS Safari has always had a quirk with how focus is handled, since you can focus an input just fine, but only a real touch event can bring up the keyboard, making it useless for that input to be focussed if the keyboard isn't displayed. 

In previous iOS versions, you _could_ focus an input as you are doing, if the keyboard is already open. I'm afraid the underlying bug is going to be in the version of Safari installed on iOS 13 devices, but we will do some investigation and see if a workaround can be provided, or if we need to submit a bug report to Apple.

In addition, it looks like Safari in iOS 13 also broke our mobile tabbing code. We'll have to look into that as well.

**Comment by wiranoid** [comment]
Any update on this?

**Comment by crookedneighbor** [comment]
No updates at this time. I'm afraid there's not much we can do if iOS Safari is preventing the focus of inputs inside of iframes.

**Comment by crookedneighbor** [comment]
Apple has updated desktop safari to prevent focussing as well. See https://github.com/braintree/braintree-web/issues/490#issuecomment-658362948

I believe it's the same underlying issue, that Apple wants to prevent inputs from being focussed without a user action, but since the user action triggers a post message to the iframe, there's no way to connect the focus request in the iframe to the original user action. 😞 

We are reaching out to Apple about this to see if there's anything that can be done.

**Comment by dboshkoff** [comment]
Hi @crookedneighbor  just wanted to follow up on this issue. We are seeing that when a user clicks on a label for a hosted fields, focus is not automatically sent into the input field. We are using the floating label design pattern, so this bug actually prevents users from selecting a hosted field unless the user clicks on the part of the field not covered by the label.  Do you think this is the same underlying issue focus issue?

I am able reproduce this behavior in Braintree's example codepen: https://codepen.io/braintree/full/mPgdPN.

**Comment by crookedneighbor** [comment]
It is.

**Comment by crookedneighbor** [comment]
After almost 2 years of this, we've finally found a fix! Released in v3.78.2

**Comment by JuanGongora** [comment]
I'm still seeing this issue when the braintree credit card payment form is being loaded up through an iframe on an iphone 11
